src/core/leaf.coffee

```diff
@@ -41,5 +41,37 @@ class Leaf extends LinkedList.Node
         @node = textNode
     @length = @text.length
 
+  findAdjacentLeaves: (offset, formats = {}) ->
+    leaves = [this]
+
+    # Collect previous leaves + update offset from click-point
+    prev = @prev;
+    while prev?
+      if prev.hasFormats(formats)
+        offset += prev.length
+        leaves = [prev].concat(leaves)
+        prev = prev.prev
+      else
+        prev = null
+
+    # Collect next leaves
+    next = @next;
+    while next?
+      if next.hasFormats(formats)
+        leaves = leaves.concat([next])
+        next = next.next
+      else
+        next = null
+
+    return [leaves, offset]
+
+  hasFormats: (formats) ->
+    leafHasFormats = true
+    for formatKey of formats
+      if not @formats.hasOwnProperty(formatKey) or @formats[formatKey] != formats[formatKey]
```

Comment:
I don't think the `hasOwnProperty` check is necessary here, and kindof hurts scannability.